### DISCUSSION TITLE
How to monitor Hasura performance ?

### DISCUSSION THREAD

**Discussion by Aarbel** [hasura/graphql-engine#8647] [Question]
Hi Hasura team and community.

We are using Sentry.io performance monitoring on both front-end (React) and Back-end (Apollo).

Problem: Hasura has no integrations with Sentry and it's like a blackbox hard to monitor, especially about performance metrics.

What are the best tools / methodologies to monitor Hasura internal performance ? 

Thanks a lot

**Comment by adas98012** [comment]
Hi,
     Thanks for reaching out. Hasura currently supports the following external tools/services to monitor performance logs/metrics (Reference Doc: https://hasura.io/docs/latest/graphql/cloud/metrics/integrations/index/)
     
Thanks,

**Reply by 1fexd** [reply]
How do I monitor a self-hosted Hasura instance?

**Reply by Aarbel** [reply]
still same question as @1fexd @adas98012

**Reply by adas98012** [reply]
@1fexd @Aarbel 
Observability features are available for self-hosted enterprise. Here is the [doc](https://hasura.io/docs/latest/enterprise/overview/) that spells out the features in some detail.

**Reply by Edmartt** [reply]
very funny @adas98012 at least let the people integrate their own choices.

**Reply by adas98012** [reply]
@Edmartt 
   We have [enterprise trial](https://hasura.io/docs/latest/enterprise/try-hasura-enterprise-edition/) available if you want to explore your observability tools of choice, with Hasura GraphQL engine.

**Reply by Aarbel** [reply]
I'm still surprised that it's not easy to add any simple monitoring tool like sentry on a basic hasura architecture.
Monitoring Hasura self hosted is a big pain unless paying thousands of $ per month for entreprise plan.

**Reply by kokroo** [reply]
Making something as basic as observability a paid feature, on software that has contributions from people around the world is just laughable.

Hasura is slowly nerfing the self-hosted version, even though there are numerous code contributions done by everyday users for free.

**Reply by Aarbel** [reply]
@adas98012 

Would we have some ways with Hasura V3 to include sentry to monitoring what happens inside Hasura self hoested ? 

Thanks a lot !

**Reply by Aarbel** [reply]
@adas98012 do you have any news about performance monitoring without having to pay hasura ? (self hosted)

**Reply by Aarbel** [reply]
@adas98012 i just saw that this performance tracing can begin with Hasura Query Tags, but this is also an only entreprise plan.
Looks like you are locking all the basic features to force people to pay, even if they use open source self hosted.

Can you please unlock at least this feature (Hasura Query Tags) for self hosted (non entreprise plans) ? 

Cf https://github.com/hasura/graphql-engine/issues/10224

**Reply by Edmartt** [reply]
Stop following this project. They are just taking advantage of people for making everything private with their dual license like the new Redis.

**Comment by hgiasac** [comment]
FYI some of Hasura cloud monitoring services collect logs and ingest metrics via ETL tools. That means it is also possible with Hasura OSS as well. Most observability services such as Datadog, HoneyComb, and Google Cloud Logging support transforming metrics from logs. 

If you want to export Prometheus, or OpenTelemetry metrics from logs, [vector](https://vector.dev/) and [OpenTelemetry Collector](https://github.com/open-telemetry/opentelemetry-collector-contrib) can help you. There are many alternatives there.

See [GraphQL Engine logs](https://hasura.io/docs/latest/deployment/logging/#http-log-structure) docs for log structure examples. They are in JSON format and easy to be parsed. 

If you want a native OpenTelemetry exporter, let's try the Hasura v3 ecosystem. The Hasura v3 OSS version and data connectors natively support OpenTelemetry. You can fork, contribute, or customize your connectors as you desire with any language. Rust is also easier to understand than Haskell.

https://hasura.io/docs/3.0/observability/traces
